sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -4488,4 +4490,32 @@ private static List<FeedRange> toFeedRanges(
     private static FeedRange toFeedRange(PartitionKeyRange pkRange) {
         return new FeedRangeEpkImpl(pkRange.toRange());
     }
+
+    /**
+     * This class generates fast UUID instances using a pseudo random number generator.
+     * It should be used when speed is more important than security.
+     * */
+    class FastRandomUUIDFactory {
+        long mostSigBits;
+        long leastSigBits;
+
+        public FastRandomUUIDFactory() {
+            mostSigBits = Thread.currentThread().getId();
+            leastSigBits = System.currentTimeMillis();
+        }
+
+        public synchronized UUID get() {
```

Comment:
@alzimmermsft can we re-use CoreUtils.randomUUID()?
We are looking for thread safe implementation which doesn't require synchronization. 
@mkrueger - we should not use synchronization here, as it will have the same locking issues which we are seeing with current java implementation

Reply:
Yeah, feel free to copy it, it's in unreleased code right now and will be available next release

Reply:
@mkrueger can we please utilize the same UUID library / class from azure-core what Alan has recommended above, thanks!

cc @FabianMeiswinkel